switchboard/base.py

```diff
@@ -32,12 +32,17 @@ def __init__(self, cache=None, timeout=30):
     def __getitem__(self, key):
         self._populate()
         try:
-            return self._cache[key]
+            value = self._cache[key]
         except KeyError:
             value = self.get_default(key)
             if value is NoValue:
                 raise
-            return value
+        except (TypeError, AttributeError):
```

Comment:
Yes:

~~~~
>>> foo = None
>>> foo['bar']
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'NoneType' object has no attribute '__getitem__'
>>> foo.get('bar')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'get'
~~~~

Attempting to access a key using the square bracket notation when the dictionary is None raises a `TypeError`.